code/game/objects/items/storage/belt.dm

```diff
@@ -635,31 +636,28 @@
 	SIGNAL_HANDLER
 	COOLDOWN_START(src, resheath_cooldown, 10 SECONDS)
 
-/datum/action/cooldown/spell/pointed/blade_counter
+/datum/action/innate/blade_counter
 	name = "Counterattack"
 	desc = "Anticipate an enemy's attack and strike back with your sheathed blade."
+	button_icon = 'icons/mob/actions/actions_spells.dmi'
 	button_icon_state = "declaration"
 	ranged_mousepointer = 'icons/effects/mouse_pointers/honorbound.dmi'
 
-	cooldown_time = 60 SECONDS
-	spell_requirements = NONE
-	antimagic_flags = NONE
-	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_LYING
+	enable_text = "You prepare to counterattack a target..."
+	disable_text = "You relax your stance."
 
-	active_msg = "You prepare to counterattack a target..."
-	deactive_msg = "You relax your stance."
-	cast_range = 2
+	click_action = TRUE
 
 	var/datum/weakref/blade_sheath_ref
```

Comment:
No need to track the sheath, innate actions have their source item stored in their ``target`` variable